### DISCUSSION TITLE
How to make a table with a fixed header and scrollable body?

### DISCUSSION THREAD

**Discussion by pidanou** [huntabyte/shadcn-svelte#1334] [Help]
Hi, I just can't manage to create a table with a fixed header and scrollable body. Either everything is fixed or the header also moves.
What I did is: `overflow-hidden` the table, and `overflow-auto` the body. Am I missing something?
Here is my code:
```
<div class="flex flex-col gap-y-2 overflow-hidden">
  <Button class="w-min">Generate</Button>
  <Table.Root class="overflow-hidden">
    <Table.Header>
      <Table.Row>
        <Table.Head
          ><Checkbox
            onCheckedChange={(v) => {
              if (v) {
                addAll();
              } else {
                removeAll();
              }
            }}
          /></Table.Head
        >
        <Table.Head>id</Table.Head>
        <Table.Head>Libelle</Table.Head>
      </Table.Row>
    </Table.Header>
    <Table.Body class="overflow-y-auto">
      {#each zones as zone}
        {@const checked = selected.includes(zone.id)}
        <Table.Row>
          <Table.Cell
            ><Checkbox
              {checked}
              onCheckedChange={(v) => {
                if (v) {
                  addItem(zone.id);
                } else {
                  removeItem(zone.id);
                }
              }}
            /></Table.Cell
          >
          <Table.Cell class="font-medium">{zone.id}</Table.Cell>
          <Table.Cell>{zone.libelle}</Table.Cell>
        </Table.Row>
      {/each}
    </Table.Body>
  </Table.Root>
</div>
```

**Comment by kaechele** [comment]
I was trying to figure this out as well.

For the table to scroll, the parent element needs to have a size that actually makes the container overflow.  `<Table.Root>` introduces another `<div>` around the table that you cannot style directly. Therefor you need to use the parent element of your `<Table.Root>` to set the height of the table. For example if you want `h-full` this works for me:

```
<div class="rounded-md border *:h-full overflow-clip">
	<Table.Root>
		<Table.Header class="sticky top-0 bg-background">
```

Note the `*:h-full` to style the child element. The `overflow-clip` is just to retain the rounded borders on the now `sticky` header. If you don't want a rounded box around the table you can leave it out.
